osquery/core/error.cpp

```diff
@@ -0,0 +1,54 @@
+//
+//  error.cpp
+//  gmock
+//
+//  Created by Max Kareta on 5/15/18.
+//
+
+#include <include/osquery/error.h>
+#include <include/osquery/status.h>
+
+namespace osquery {
+
+Error::Error(std::string domain,
+             int error_code,
+             std::string message,
+             std::shared_ptr<Error> underlying_error)
+    : domain_(domain),
+      errorCode_(error_code),
+      message_(message),
+      underlyingError_(underlying_error) {}
```

Comment:
It would be better to avoid useless access to ref counter for stared_ptr. Just use `std::move` here.